python/mlx/nn/losses.py

```diff
@@ -283,3 +283,83 @@ def _reduce(loss: mx.array, reduction: str = "none"):
         return loss
     else:
         raise ValueError("Invalid reduction. Must be 'none', 'mean', or 'sum'.")
+
+
+def hinge_loss(
+    inputs: mx.array, targets: mx.array, reduction: str = "none"
+) -> mx.array:
+    """
+    Computes the hinge loss between inputs and targets.
+
+    .. math::
+
+       \text{hinge}(y, y_{\text{pred}}) = \max(0, 1 - y \cdot y_{\text{pred}})
+
+
+    Args:
+        inputs (array): The predicted values.
+        targets (array): The target values.
+        reduction (str, optional): Specifies the reduction to apply to the output:
+          ``'none'`` | ``'mean'`` | ``'sum'``. Default: ``'none'``.
+
+    Returns:
+        array: The computed hinge loss.
+    """
+    loss = mx.maximum(1 - inputs * targets, 0)
+
+    return _reduce(loss, reduction)
+
+
+def huber_loss(
+    inputs: mx.array, targets: mx.array, delta: float = 1.0, reduction: str = "none"
+) -> mx.array:
+    """
+    Computes the Huber loss between inputs and targets.
+
+    .. math::
+
+       L_{\delta}(a) = \left\{ \begin{array}{ll}\frac{1}{2} a^2 & \text{for } |a| \leq \delta, \\\delta \left( |a| - \frac{1}{2} \delta \right) & \text{otherwise.}\end{array} \right.
```

Comment:
Could you maybe format the latex in multiple lines? Also, I didn't render it but I think there is a brace missing after the last `\right`.

Reply:
Hey @angeloskath that's the rendered. I think it's ok

![Screenshot 2023-12-21 at 09 42 19](https://github.com/ml-explore/mlx/assets/37878833/0a655d70-60db-4cac-965b-44d8fdeecf2d)

Reply:
My bad, it looks great. Too used to seeing `\left( \right)` I jumped the gun :-)